Data/Vector/Mutable.hs

```diff
@@ -428,3 +431,23 @@ unsafeMove = G.unsafeMove
 nextPermutation :: (PrimMonad m,Ord e) => MVector (PrimState m) e -> m Bool
 {-# INLINE nextPermutation #-}
 nextPermutation = G.nextPermutation
+
+-- Conversions - Arrays
+-- -----------------------------
+
+-- | /O(n)/ Make a copy of a mutable array to a new mutable vector.
+fromMutableArray :: PrimMonad m => MutableArray (PrimState m) a -> m (MVector (PrimState m) a)
+{-# INLINE fromMutableArray #-}
+fromMutableArray marr = do
```

Comment:
I don't really know how I feel about this function. Whenever I've wanted `fromMutableArray` before, I've wanted it to alias the original array, not to make a copy. I think I would prefer something like:

    fromMutableArray :: MutableArray (PrimState m) a -> Int -> Int -> MVector (PrimState m) a

Which explicitly accepts the offsets (and is just a wrapper for the data constructor). Such an interface is more powerful, since it would allow me to temporarily or permanently "step up" into an `MVector` during a computation if there is some machinery there that I happen to need. It's possible to provide both interfaces though.

Reply:
I disagree on this one. `MVector` is exported, so the function you are talking about would be just a synonym for the constructor. If a user wants to manually deal with dangerous indices and offsets then we won't prevent him, but we should not promote it by creating a separate function for such functionality.